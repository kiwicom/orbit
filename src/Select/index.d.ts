// @flow
// Type definitions for @kiwicom/orbit-components
// Project: http://github.com/kiwicom/orbit-components

import * as React from "react";

import * as Common from "../common/common";

declare module "@kiwicom/orbit-components/lib/Select";

// InputEvent
type Event = Common.Event<React.SyntheticEvent<HTMLSelectElement>>;

interface Option {
  readonly value: string | number;
  readonly label?: string;
  readonly disabled?: boolean;
}

export interface Props extends Common.Global, Common.Ref, Common.SpaceAfter, Common.DataAttrs {
  readonly id?: string;
  readonly required?: boolean;
  readonly size?: Common.InputSize;
  readonly label?: Common.Translation;
  readonly placeholder?: Common.Translation;
  readonly value?: string | number;
  readonly disabled?: boolean;
  readonly name?: string;
  readonly error?: React.ReactNode;
  readonly help?: React.ReactNode;
  readonly tabIndex?: string;
  readonly onChange?: Event;
  readonly onFocus?: Event;
  readonly onBlur?: Event;
  readonly options: Option[];
  readonly prefix?: React.ReactNode;
  readonly customValueText?: Common.Translation;
}

declare const Select: React.RefForwardingComponent<Props, HTMLInputElement>;
export { Select, Select as default };
