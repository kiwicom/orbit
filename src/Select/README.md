# Select
To implement Select component into your project you'll need to add the import:
```jsx
import Select from "@kiwicom/orbit-components/lib/Select";
```
After adding import into your project you can use it simply like:
```jsx
<Select options={Option} />
```
## Props
Table below contains all types of the props available in the Select component.

| Name        | Type                              | Default    | Description                      |
| :---------- | :-------------------------------- | :--------- | :------------------------------- |
| dataTest    | `string`                          |            | Optional prop for testing purposes.
| disabled    | `boolean`                         | `false`    | If `true`, the Select will be disabled.
| error       | `React.Node`                      |            | The error message for the Select. [See Functional specs](#functional-specs)
| help        | `React.Node`                      |            | The help message for the Select.
| label       | `Translation`                     |            | The label for the Select.
| name        | `string`                          |            | The name for the Select.
| onBlur      | `event => void \| Promise`        |            | Function for handling onBlur event.
| onChange    | `event => void \| Promise`        |            | Function for handling onChange event.
| onFocus     | `event => void \| Promise`        |            | Function for handling onFocus event.
| **options** | [`Option[]`](#option)             |            | The content of the Select, passed as array of objects.
| placeholder | `Translation`                     |            | The placeholder for the Select. 
| prefix      | `React.Node`                      |            | The prefix component for the Select. [See Functional specs](#functional-specs)
| ref         | `func`                            |            | Prop for forwarded ref of the Select. [See Functional specs](#functional-specs)
| size        | [`enum`](#enum)                   | `"normal"` | The size of the Select.
| spaceAfter  | `enum`                            |            | Additional `margin-bottom` after component. [See this docs](https://github.com/kiwicom/orbit-components/tree/master/src/common/getSpacingToken)
| tabIndex    | `string`                          |            | Specifies the tab order of an element
| value       | `string`                          | `""`       | The value of the Select.

## Option
Table below contains all types of the props available for object in Option array.

| Name         | Type               | Description                      |
| :----------- | :----------------- | :------------------------------- |
| **value**    | `string \| number` | The value of the Option.
| label        | `string`           | The label for the Option.
| disabled     | `boolean`          | If `true`, the Option will be disabled.
  
### enum

| size       |
| :--------- |
| `"small"`  |
| `"normal"` |

## Functional specs
* The `error` prop overwrites the `help` prop, due to higher priority.

* `ref` can be used for example auto-focus the elements immediately after render.
```jsx
class Component extends React.PureComponent<Props> {

  componentDidMount() {
    this.ref.current && this.ref.current.focus();
  }

  ref: { current: React$ElementRef<*> | null } = React.createRef();

  render() {
    return (
      <Select ref={this.ref} />
    )
  }
}
```


