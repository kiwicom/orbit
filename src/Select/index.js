// @flow
import * as React from "react";
import styled from "styled-components";

import defaultTheme from "../defaultTheme";
import FormLabel from "../FormLabel";
import ChevronDown from "../icons/ChevronDown";
import FormFeedback from "../FormFeedback";
import TYPE_OPTIONS from "../FormFeedback/consts";
import SIZE_OPTIONS from "./consts";
import type { Ref } from "../common/common.js.flow";
import { right, rtlSpacing } from "../utils/rtl";
import getSpacingToken from "../common/getSpacingToken";

import type { Props } from "./index";

const getSelectSize = ({ theme, size }) => {
  const tokens = {
    [SIZE_OPTIONS.SMALL]: theme.orbit.heightInputSmall,
    [SIZE_OPTIONS.NORMAL]: theme.orbit.heightInputNormal,
  };
  return tokens[size];
};

const Label = styled.label`
  position: relative;
  display: block;
  width: 100%;
  margin-bottom: ${getSpacingToken};
`;

Label.defaultProps = {
  theme: defaultTheme,
};

const StyledSelect = styled(
  // $FlowExpected
  React.forwardRef(
    (
      { className, dataTest, children, value, disabled, name, tabIndex, onChange, onFocus, onBlur },
      ref,
    ) => (
      <select
        data-test={dataTest}
        value={value}
        className={className}
        onChange={onChange}
        onFocus={onFocus}
        onBlur={onBlur}
        disabled={disabled}
        name={name}
        tabIndex={tabIndex}
        ref={ref}
      >
        {children}
      </select>
    ),
  ),
)`
  appearance: none;
  background: ${({ theme }) => theme.orbit.backgroundInput};
  border-radius: ${({ theme }) => theme.orbit.borderRadiusNormal};
  cursor: pointer;
  color: ${({ theme, filled }) =>
    filled ? theme.orbit.colorTextInput : theme.orbit.colorPlaceholderInput};
  font-family: ${({ theme }) => theme.orbit.fontFamily};
  font-size: ${({ theme, size }) =>
    size === SIZE_OPTIONS.SMALL ? theme.orbit.fontSizeInputSmall : theme.orbit.fontSizeInputNormal};
  height: ${getSelectSize};
  padding: ${({ theme, size, prefix }) =>
    rtlSpacing(
      (size === SIZE_OPTIONS.SMALL &&
        !prefix &&
        `0 ${theme.orbit.spaceXLarge} 0 ${theme.orbit.spaceSmall}`) ||
        (size === SIZE_OPTIONS.SMALL &&
          prefix &&
          `0 ${theme.orbit.spaceXLarge} 0 ${theme.orbit.paddingLeftSelectPrefix}`) ||
        (prefix && `0 ${theme.orbit.spaceXXLarge} 0 ${theme.orbit.paddingLeftSelectPrefix}`) ||
        `0 ${theme.orbit.spaceXXLarge} 0 ${theme.orbit.spaceSmall}`,
    )};
  outline: none;
  width: 100%;
  transition: box-shadow ${({ theme }) => theme.orbit.durationFast} ease-in-out;
  z-index: 2;

  > option {
    color: ${({ theme }) => theme.orbit.colorTextInput};
  }

  &::-ms-expand {
    display: none;
  }

  // IE Bug fix: highlighted background color and color of text
  &::-ms-value {
    background: transparent;
    color: ${({ theme, filled }) =>
      filled ? theme.orbit.colorTextInput : theme.orbit.colorPlaceholderInput};
  }

  /* Based on state of select */
  border: 0;
  box-shadow: inset 0 0 0
    ${({ theme, error }) =>
      `${theme.orbit.borderWidthInput} ${
        error ? theme.orbit.borderColorInputError : theme.orbit.borderColorInput
      }`};

  &:hover {
    box-shadow: inset 0 0 0
      ${({ theme, error }) =>
        `${theme.orbit.borderWidthInput} ${
          error ? theme.orbit.borderColorInputErrorHover : theme.orbit.borderColorInputHover
        }`};
  }

  &:focus {
    box-shadow: inset 0 0 0
      ${({ theme }) => `${theme.orbit.borderWidthInputFocus} ${theme.orbit.borderColorInputFocus}`};
  }

  &:disabled {
    color: ${({ theme }) => theme.orbit.colorTextInputDisabled};
    background: ${({ theme }) => theme.orbit.backgroundInputDisabled};
    cursor: default;

    &:hover {
      box-shadow: inset 0 0 0 1px ${({ theme }) => theme.orbit.borderColorInput};
    }
  }
`;

StyledSelect.defaultProps = {
  theme: defaultTheme,
};

export const SelectContainer = styled(({ className, children }) => (
  <div className={className}>{children}</div>
))`
  position: relative;
  display: flex;
  align-items: center;
  background: ${({ theme }) => theme.orbit.backgroundInput};
  width: 100%;
  box-sizing: border-box;
  cursor: pointer;
`;

SelectContainer.defaultProps = {
  theme: defaultTheme,
};

const SelectPrefix = styled(({ className, children }) => (
  <div className={className}>{children}</div>
))`
  display: flex;
  align-items: center;
  position: absolute;
  padding: ${({ theme }) => `0 ${theme.orbit.spaceSmall}`};
  pointer-events: none;
  z-index: 3;
  top: 0;
  height: ${getSelectSize};
`;

SelectPrefix.defaultProps = {
  theme: defaultTheme,
};

const SelectSuffix = styled(({ children, className }) => (
  <div className={className}>{children}</div>
))`
  position: absolute;
  display: flex;
  justify-content: center;
  align-items: center;
  top: 0;
  ${right}: ${({ theme }) => theme.orbit.spaceXSmall};
  color: ${({ theme, disabled }) =>
    disabled ? theme.orbit.colorTextInputDisabled : theme.orbit.colorTextInput};
  pointer-events: none;
  z-index: 3;
  height: 100%;

  & > * {
    width: ${({ theme, size }) => size === SIZE_OPTIONS.SMALL && theme.orbit.widthIconSmall};
    height: ${({ theme, size }) => size === SIZE_OPTIONS.SMALL && theme.orbit.heightIconSmall};
    margin-bottom: ${({ size, theme }) => size === SIZE_OPTIONS.SMALL && theme.orbit.spaceXXXSmall};
  }
`;

SelectSuffix.defaultProps = {
  theme: defaultTheme,
};

// $FlowExpected
const Select = React.forwardRef((props: Props, ref: Ref) => {
  const {
    size = SIZE_OPTIONS.NORMAL,
    label,
    placeholder,
    value,
    disabled = false,
    error,
    help,
    name,
    onChange,
    onBlur,
    onFocus,
    options,
    tabIndex,
    dataTest,
    prefix,
    spaceAfter,
  } = props;
  const filled = !(value == null || value === "");
  return (
    <Label spaceAfter={spaceAfter}>
      {label && (
        <FormLabel filled={filled} disabled={disabled}>
          {label}
        </FormLabel>
      )}
      <SelectContainer disabled={disabled}>
        {prefix && (
          <SelectPrefix prefix={prefix} size={size}>
            {prefix}
          </SelectPrefix>
        )}
        <StyledSelect
          dataTest={dataTest}
          size={size}
          disabled={disabled}
          error={error}
          value={value == null ? "" : value}
          prefix={prefix}
          name={name}
          onFocus={onFocus}
          onBlur={onBlur}
          onChange={onChange}
          filled={filled}
          tabIndex={tabIndex}
          ref={ref}
        >
          {placeholder && (
            <option label={placeholder} value="">
              {placeholder}
            </option>
          )}
          {options.map(option => (
            <option key={`option-${option.value}`} value={option.value} disabled={option.disabled}>
              {option.label}
            </option>
          ))}
        </StyledSelect>
        <SelectSuffix size={size} disabled={disabled}>
          <ChevronDown />
        </SelectSuffix>
      </SelectContainer>
      {help && !error && <FormFeedback type={TYPE_OPTIONS.HELP}>{help}</FormFeedback>}
      {error && <FormFeedback type={TYPE_OPTIONS.ERROR}>{error}</FormFeedback>}
    </Label>
  );
});

// otherwise Unknown in storybook
Select.displayName = "Select";

export default Select;
