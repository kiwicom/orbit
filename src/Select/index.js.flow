// @flow
/*
  DOCUMENTATION: https://orbit.kiwi/components/select/
*/
import type { ReactComponentStyled } from "styled-components";

import type {
  Globals,
  Ref,
  Translation,
  TranslationString,
  DataAttrs,
} from "../common/common.js.flow";
import type { spaceAfter } from "../common/getSpacingToken/index";

type Option = {|
  +value: string | number,
  +label?: string,
  +disabled?: boolean,
|};

type Size = "small" | "normal";

export type Props = {|
  ...Globals,
  ...Ref,
  ...spaceAfter,
  ...DataAttrs,
  +id?: string,
  +required?: boolean,
  +size?: Size,
  +label?: Translation,
  +placeholder?: TranslationString,
  +value?: string | number,
  +disabled?: boolean,
  +name?: string,
  +error?: React$Node,
  +help?: React$Node,
  tabIndex?: string,
  +onChange?: (ev: SyntheticInputEvent<HTMLSelectElement>) => void | Promise<any>,
  +onFocus?: (ev: SyntheticInputEvent<HTMLSelectElement>) => void | Promise<any>,
  +onBlur?: (ev: SyntheticInputEvent<HTMLSelectElement>) => void | Promise<any>,
  +options: Option[],
  +prefix?: React$Node,
  +customValueText?: Translation,
|};

declare export var SelectContainer: ReactComponentStyled<any>;

declare export default React$ComponentType<Props>;
