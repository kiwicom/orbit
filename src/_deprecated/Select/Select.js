// @flow
import * as React from "react";
import css from "styled-jsx/css";
import ChevronDown from "react-icons/lib/fa/chevron-down";
import classnames from "classnames";

import Typography from "../Typography/Typography";
import { fontColors } from "../../constants";
import FieldFeedback from "../FieldFeedback/FieldFeedback";

const WHITE_COLOR = "white";

const HASERROR_CLASS = "hasError";
const UNSELECTED_CLASS = "UnSelected";

type Option = {
  value: string | number,
  label?: string,
  disabled?: boolean,
  visible?: boolean, // eslint-disable-line react/no-unused-prop-types
};
type Props = {
  disabled?: boolean,
  error?: string,
  help?: string,
  label?: string,
  onChange: (SyntheticInputEvent<HTMLSelectElement>) => any,
  options: Array<any>,
  placeholder?: string,
  required?: boolean,
  value: string,
};
type State = {
  value: string,
};

const style = css`
  label {
    font-family: Roboto;
    line-height: 1.43;
    position: relative;
    color: ${fontColors.primary};
  }
  select {
    font-size: 16px;
    border-radius: 3px;
    background-color: ${WHITE_COLOR};
    border: solid 1px ${fontColors.input};
    line-height: 1.25;
    padding: 12px 16px;
    color: ${fontColors.primary};
    appearance: none;
    padding-right: 3em;
    width: 100%;
  }
  select.${UNSELECTED_CLASS} {
    color: ${fontColors.secondary};
  }
  /*for IE10*/
  select::-ms-expand {
    display: none;
  }
  select.${HASERROR_CLASS} {
    border-color: ${fontColors.error};
  }
  option[value=""][disabled] {
    display: none;
  }
  .dropdown-icon {
    position: absolute;
    right: 1em;
    top: 0px;
    pointer-events: none;
  }
`;

class Select extends React.Component<Props, State> {
  state = {
    value: this.props.value,
  };
  handleChange = (e: SyntheticInputEvent<HTMLSelectElement>) => {
    this.setState({ value: e.target.value });
    this.props.onChange(e);
  };
  renderOption = ({ value, label, disabled, visible = true }: Option) => {
    if (visible) {
      return (
        <option key={`option-${value}`} value={value} label={label || value} disabled={!!disabled}>
          {label || value}
        </option>
      );
    }
    return null;
  };
  render() {
    const { disabled, label, options, error, help, placeholder, required = true } = this.props;
    const { value } = this.state;
    return (
      <div>
        <label>
          {label && (
            <div>
              <Typography size="normal" type={value ? "input" : "primary"}>
                {label}
              </Typography>
            </div>
          )}
          <select
            disabled={disabled}
            value={value}
            required={!!required}
            onChange={this.handleChange}
            className={classnames(error && HASERROR_CLASS, !value && UNSELECTED_CLASS)}
          >
            {
              <option disabled value="" label={placeholder} className="placeholder">
                {placeholder}
              </option>
            }
            {options.map(this.renderOption)}
          </select>
          <span className="dropdown-icon">
            <ChevronDown fill={fontColors.primary} height={20} />
          </span>
        </label>
        <FieldFeedback error={error} help={help} />
        <style jsx>{style}</style>
      </div>
    );
  }
}

export default Select;
