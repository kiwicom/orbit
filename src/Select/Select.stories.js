// @flow
import * as React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import { object, select, text, boolean } from "@storybook/addon-knobs";

import Airplane from "../icons/Airplane";
import SIZE_OPTIONS from "./consts";
import CountryFlag from "../CountryFlag";
import { CODES } from "../CountryFlag/consts";
import RenderInRtl from "../utils/rtl/RenderInRtl";
import SPACINGS_AFTER from "../common/getSpacingToken/consts";

import Select from "./index";

const objectOptions = [
  { value: 0, label: "Zero-th item" },
  { value: 1, label: "First item" },
  { value: 2, label: "Second item" },
  { value: 3, label: "Third item" },
];

storiesOf("Select", module)
  .add("Default", () => <Select options={objectOptions} onChange={action("onChange")} />, {
    info:
      "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
  })
  .add(
    "With prefix",
    () => (
      <Select
        label="Select box (with prefix)"
        options={objectOptions}
        onChange={action("onChange")}
        prefix={<Airplane color="secondary" />}
      />
    ),
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "With CountryFlag prefix",
    () => {
      const code = select("Code", Object.values(CODES), CODES.ANYWHERE);
      return (
        <Select
          label="Select box (with prefix)"
          options={objectOptions}
          onChange={action("onChange")}
          prefix={<CountryFlag code={code} />}
        />
      );
    },
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "With placeholder",
    () => {
      const placeholder = text("Placeholder", "Select value from list");
      return (
        <Select
          label="Select box (with placeholder)"
          placeholder={placeholder}
          options={objectOptions}
          onChange={action("onChange")}
        />
      );
    },
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "With help message",
    () => (
      <Select
        label="Select box (with help text)"
        options={objectOptions}
        help="Most common choice is Booking cancellation"
        onChange={action("onChange")}
      />
    ),
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "With error message",
    () => (
      <Select
        label="Select box (with error text)"
        options={objectOptions}
        error={<div>You need to select some value.</div>}
        onChange={action("onChange")}
      />
    ),
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "With small size",
    () => (
      <Select
        label="Select box (small size)"
        size="small"
        options={objectOptions}
        onChange={action("onChange")}
      />
    ),
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "Playground",
    () => {
      const placeholder = text("Placeholder", "Select value from list");
      const size = select("Size", Object.values(SIZE_OPTIONS), SIZE_OPTIONS.NORMAL);
      const disabled = boolean("Disabled", false);
      const name = text("Name", "name");
      const option = object("Options", objectOptions);
      const value = select("Value", [undefined].concat(...objectOptions.map(opt => opt.value)));
      const dataTest = text("dataTest", "test");
      const spaceAfter = select("spaceAfter", [null, ...Object.values(SPACINGS_AFTER)]);

      return (
        <Select
          placeholder={placeholder}
          size={size}
          options={option}
          disabled={disabled}
          name={name}
          label={text("Label")}
          onChange={action("onChange")}
          onBlur={action("onBlur")}
          onFocus={action("onFocus")}
          dataTest={dataTest}
          value={value}
          spaceAfter={spaceAfter}
        />
      );
    },
    {
      info:
        "Selects are used for showing content hierarchy and are important for improving the reading experience for our users. Visit Orbit.Kiwi for more detailed guidelines.",
    },
  )
  .add(
    "RTL",
    () => (
      <RenderInRtl>
        <Select placeholder="My placeholder" options={objectOptions} label="My label" />
      </RenderInRtl>
    ),
    {
      info: "This is a preview of this component in RTL setup.",
    },
  );
