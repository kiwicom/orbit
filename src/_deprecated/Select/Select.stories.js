// @flow

import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";

import Select from "./index";

const objectOptions = [
  { value: 1, label: "One" },
  { value: 2, label: "Two" },
  { value: 3, label: "Three" },
  { value: 4, label: "Four" },
];

const hiddenOptions = [
  { value: "one", label: "One" },
  { value: "two", label: "Two" },
  { value: "three", label: "Three" },
  { visible: false, value: "hidden-four", label: "Hidden Four" },
];
const disabledOptions = [
  { value: "one", label: "One" },
  { value: "two", label: "Two" },
  { value: "three", label: "Three" },
  { disabled: true, value: "hidden-four", label: "Hidden Four" },
];

storiesOf("Select", module)
  .add("Default", () => <Select value="" options={objectOptions} onChange={action("onChange")} />)
  .add("Default with placeholder", () => (
    <Select
      value=""
      placeholder="MyPlaceHolder"
      options={objectOptions}
      onChange={action("onChange")}
    />
  ))
  .add("Hidden 'Four' option", () => (
    <Select value="" options={hiddenOptions} onChange={action("onChange")} />
  ))
  .add("Disabled 'Four' option", () => (
    <Select value="" options={disabledOptions} onChange={action("onChange")} />
  ))
  .add("With error feedback", () => (
    <Select
      value=""
      options={objectOptions}
      error="Error message here"
      onChange={action("onChange")}
    />
  ))
  .add("With help feedback", () => (
    <Select
      value=""
      options={objectOptions}
      help="A helpful message here"
      onChange={action("onChange")}
    />
  ))
  .add("With label NOT filled", () => (
    <Select label="Choose number" value="" options={objectOptions} onChange={action("onChange")} />
  ))
  .add("With label with help", () => (
    <Select
      label="Choose number"
      help="Helpful feedback"
      value=""
      options={objectOptions}
      onChange={action("onChange")}
    />
  ))
  .add("With label filled", () => (
    <Select label="Choose number" value="2" options={objectOptions} onChange={action("onChange")} />
  ));
