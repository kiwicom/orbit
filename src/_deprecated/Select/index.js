// @flow
import Select from "./Select";
import DeprecatedComponent from "../DeprecatedComponent";

export default DeprecatedComponent(
  Select,
  "Select is deprecated and will be removed in next major release",
);
