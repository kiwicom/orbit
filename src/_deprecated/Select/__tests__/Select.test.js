// @flow

import * as React from "react";
import { shallow } from "enzyme";

import Select from "../Select";

const objectOptions = [
  { value: "1", label: "One" },
  { value: "2", label: "Two" },
  { value: "3", label: "Three" },
  { value: "4", label: "Four" },
  { visible: false, value: "hidden-five", label: "Hidden Five" },
  { disabled: true, value: "disabled-six", label: "Disabled Six" },
];

const mockChange = jest.fn();
describe("Select", () => {
  const component = shallow(<Select value="1" options={objectOptions} onChange={mockChange} />);
  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });
  it("should set component state and run onChange callback", () => {
    component.find("select").simulate("change", { target: { value: objectOptions[0].value } });
    expect(component.state().value).toBe(objectOptions[0].value);
    expect(mockChange.mock.calls.length).toBe(1);
  });
});
