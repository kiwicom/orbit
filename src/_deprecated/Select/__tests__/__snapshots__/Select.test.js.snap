// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should match snapshot 1`] = `
<div
  className="jsx-2198933393"
>
  <label
    className="jsx-2198933393"
  >
    <select
      className="jsx-2198933393 "
      onChange={[Function]}
      required={true}
      value="1"
    >
      <option
        className="jsx-2198933393 placeholder"
        disabled={true}
        value=""
      />
      <option
        disabled={false}
        key="option-1"
        label="One"
        value="1"
      >
        One
      </option>
      <option
        disabled={false}
        key="option-2"
        label="Two"
        value="2"
      >
        Two
      </option>
      <option
        disabled={false}
        key="option-3"
        label="Three"
        value="3"
      >
        Three
      </option>
      <option
        disabled={false}
        key="option-4"
        label="Four"
        value="4"
      >
        Four
      </option>
      <option
        disabled={true}
        key="option-disabled-six"
        label="Disabled Six"
        value="disabled-six"
      >
        Disabled Six
      </option>
    </select>
    <span
      className="jsx-2198933393 dropdown-icon"
    >
      <FaChevronDown
        fill="#46515e"
        height={20}
      />
    </span>
  </label>
  <FieldFeedback />
  <JSXStyle
    css="label.jsx-2198933393{font-family:Roboto;line-height:1.43;position:relative;color:#46515e;}select.jsx-2198933393{font-size:16px;border-radius:3px;background-color:white;border:solid 1px #bac7d5;line-height:1.25;padding:12px 16px;color:#46515e;-webkit-appearance:none;-moz-appearance:none;appearance:none;padding-right:3em;width:100%;}select.UnSelected.jsx-2198933393{color:#7f91a8;}select.jsx-2198933393::-ms-expand{display:none;}select.hasError.jsx-2198933393{border-color:#d21c1c;}option[value=\\"\\"][disabled].jsx-2198933393{display:none;}.dropdown-icon.jsx-2198933393{position:absolute;right:1em;top:0px;pointer-events:none;}"
    styleId="2198933393"
  />
</div>
`;
